### ISSUE TITLE
incorrect `any` errors when using `await`/`then` block

### ISSUE THREAD

**Issue by DetachHead** [sveltejs/eslint-plugin-svelte3#189]
```svelte
<script>
    let foo: Promise<number[]>
</script>

{#await foo then bar}
    <button on:click={() => bar.length} /> // error: Unsafe return of an `any` typed value (@typescript-eslint/no-unsafe-return)
{/await}
```

**Comment by DetachHead** [comment]
this issue doesn't occur in [`eslint-plugin-svelte`](https://github.com/ota-meshi/eslint-plugin-svelte) which seems to be more actively maintained

- #184

**Comment by NatoBoram** [comment]
It does happen:

* https://github.com/sveltejs/eslint-plugin-svelte/issues/534

**Comment by DetachHead** [comment]
looks like that only occurs when the variables used on `#await` and `:then` are the same, so it's slightly different. the example from my OP does not trigger the error
